cmd/template-resolver/utils/templateResolver.go

```diff
@@ -172,143 +221,213 @@ func ProcessTemplate(yamlBytes []byte, hubKubeConfigPath, clusterName string) ([
 			continue
 		}
 
-		if hubResolver != nil {
-			objectDefinitionJSON, err := json.Marshal(objectDefinition)
-			if err != nil {
-				return nil, fmt.Errorf("invalid policy-templates entry at index %d: %w", i, err)
-			}
+		objectDefinition, err = processObjectTemplates(objectDefinition, resolver)
+		if err != nil {
+			return fmt.Errorf("%w (in policy-templates at index %d)", err, i)
+		}
 
-			hubTemplateResult, err := hubResolver.ResolveTemplate(
-				objectDefinitionJSON, hubTemplateCtx, &hubResolveOptions,
+		err = unstructured.SetNestedField(policyTemplate, objectDefinition, "objectDefinition")
+		if err != nil {
+			return fmt.Errorf(
+				"invalid policy-templates entry at index %d after resolving templates: %w",
+				i,
+				err,
 			)
-			if err != nil {
-				return nil, fmt.Errorf("invalid policy-templates entry at index %d: %w", i, err)
-			}
+		}
+	}
 
-			var resolvedObjectDefinition map[string]interface{}
+	err = unstructured.SetNestedSlice(policy.Object, policyTemplates, "spec", "policy-templates")
+	if err != nil {
+		return fmt.Errorf("invalid policy-templates after resolving templates: %w", err)
+	}
 
-			err = json.Unmarshal(hubTemplateResult.ResolvedJSON, &resolvedObjectDefinition)
-			if err != nil {
-				return nil, fmt.Errorf(
-					"invalid policy-templates entry at index %d after resolving templates: %w",
-					i,
-					err,
-				)
-			}
+	return nil
+}
 
-			err = unstructured.SetNestedField(policyTemplate, resolvedObjectDefinition, "objectDefinition")
-			if err != nil {
-				return nil, fmt.Errorf(
-					"invalid policy-templates entry at index %d after resolving templates: %w",
-					i,
-					err,
-				)
-			}
+// ProcessConfigPolicyTemplate takes the unmarshalled ConfigPolicy YAML as input
+// and resolves its templates
+func processConfigPolicyTemplate(
+	policy *unstructured.Unstructured,
+	resolver *templates.TemplateResolver,
+) error {
+	resolvedPolicy, err := processObjectTemplates(policy.Object, resolver)
+	if err != nil {
+		return err
+	}
 
-			objectDefinition = policyTemplate["objectDefinition"].(map[string]interface{})
-			objectDefinitionUnstructured.Object = objectDefinition
-		}
+	policy.Object = resolvedPolicy
+
+	return nil
+}
+
+// ProcessObjTemplatesRaw takes a YAML string representation and
+// resolves the object's hub and managed templates
+func processObjTemplatesRaw(
+	raw *unstructured.Unstructured,
+	resolver *templates.TemplateResolver,
+) error {
+	var rawDataList [][]byte
+
+	resolveOptions := templates.ResolveOptions{}
+
+	oTRaw, oTRawFound, _ := unstructured.NestedString(raw.Object, "object-templates-raw")
```

Comment:
The error should be account for here or alternatively, check make sure `oTRaw != ""` and you can then stop `oTRawFound` too.